pkg/cosign/remote/index.go

```diff
@@ -113,11 +113,21 @@ func UploadFiles(ref name.Reference, files []File, getMt MediaTypeGetter, remote
 		if err != nil {
 			return name.Digest{}, err
 		}
+
 		lastHash, err = img.Digest()
 		if err != nil {
 			return name.Digest{}, err
 		}
-		if err := remote.Write(ref, img, remoteOpts...); err != nil {
+
+		//  cast img to a v1.image
+		v1Img, ok := img.(v1.Image)
```

Comment:
Instead of casting to a `v1.Image`, could we do something like this up above?

```go
img, err := static.NewFile(b, static.WithLayerMediaType(mt), static.WithAnnotations(annotations))
```

Reply:
Sure!

Reply:
Hmm `static.NewFile` did not handle the `WithAnnotations` option, added it and added a test for it.